### PR TITLE
Fix daemon reload with backoff SIGINT handling

### PR DESCRIPTION
**Pull request by aleksmaus**
## What does this PR do?

Fixes SIGINT handling while doing daemon reload with backoff.

## Why is it important?

Currently SIGINT doesn't interrupt the daemon reload backoff retries upon enrollment. 
Can be reproduced for example by starting ```./elastic-agent enroll ...........``` command, and issuing Ctrl+C, agent continues running.
This fixes it. 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [x] I have added an integration test or an E2E test

### CODE REVIEW

File: internal/pkg/agent/cmd/enroll_cmd.go [hunk 1]
```diff
@@ -486,8 +490,10 @@ func (c *enrollCmd) daemonReloadWithBackoff(ctx context.Context) error {
 		lastErr = err
 
 		c.log.Errorf("Restart attempt %d failed: '%s'. Waiting for %s", attempt, err, backExp.NextWait().String())
-		backExp.Wait()
-
+		// backoff Wait returns false if context.Done()
+		if !backExp.Wait() {
```
Comment (cmacknz) [inline internal/pkg/agent/cmd/enroll_cmd.go:494]: We should add a unit test here to catch this, the simplest way I can see would be something like:

1. Setup a context that is either already cancelled or with a timeout much shorter than `daemonReloadInitBackoff`.
2. Use our mock control client to get past the call to daemonReload returning an error that is not a context error.
3. Ensure the function returns a context error faster than `daemonReloadInitBackoff`.

The mock control client:
https://github.com/elastic/elastic-agent/blob/d8b446f5a5686c99415e823edf321699fbea4b26/pkg/control/v2/client/mocks/client.go#L633-L637

Example usage:

https://github.com/elastic/elastic-agent/blob/d8b446f5a5686c99415e823edf321699fbea4b26/internal/pkg/agent/application/upgrade/upgrade_test.go#L192-L196

**Review by pchila — COMMENTED** [review]
The code looks reasonable,
I agree with @cmacknz : a unit test would be nice.

**Comment by aleksmaus** [comment]
Added unit tests. Injecting ```daemonReloadFunc``` instead. 
From the point of view of testing the  ```daemonReloadInitBackoff``` function we don't need to to inject the full mock client nor define it's behavior (it calls three different functions).

**Review by cmacknz — APPROVED** [review]
Thanks for the test!

**Pull request merged**